### PR TITLE
Selenium + login

### PR DESCRIPTION
**Pull request by KarolDenst**

### CODE REVIEW

File: backend/YouTubeV2.Application/YTContext.cs [hunk 1]
```diff
@@ -18,6 +19,52 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
             base.OnModelCreating(modelBuilder);
             modelBuilder.ApplyConfigurationsFromAssembly(typeof(UserEntityConfiguration).Assembly);
+
+            var roles = new[]
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/YTContext.cs:22]: do we need to specify ids?

File: backend/YouTubeV2.Application/YTContext.cs [hunk 2]
```diff
@@ -18,6 +19,52 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+                {
+                    Id = "b3a48a48-1a74-45da-a179-03b298bc53bc",
+                    Name = Role.Administrator,
+                    NormalizedName = Role.Administrator.ToUpper(),
+                }
+            };
+
+            modelBuilder.Entity<Role>().HasData(roles);          
+
+            var user = new User
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/YTContext.cs]: is id required to specify? Maybe change the data so that it is obvious that it is a simple user eg simpleuser@mail.com. i think if you are doing the seeding you can also add sample creator and administrator

File: frontend/playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { defineConfig, devices } from '@playwright/test';
```
Comment (Marvvvin5) [inline frontend/playwright.config.ts:1]: is it generated code? if not maybe we should delete this comments. Also idk if testing both edge and chrome is necessary since they are both on chromium as far as i know. Checking firefox/safari/opera seems to be a better idea

File: frontend/src/app/core/services/user.service.ts [hunk 4]
```diff
@@ -4,15 +4,32 @@ import { Observable, Subject } from "rxjs";
...
 import { UserForRegistrationDTO } from "../../authentication/models/user-for-registration-dto";
 import { UserDTO } from "../models/user-dto";
+import { UserForLoginDTO } from "src/app/authentication/models/user-login-dto";
 
 @Injectable({
   providedIn: 'root'
 })
 export class UserService {
   private readonly registrationPageWebAPIUrl: string = `${environment.webApiUrl}`;
+  private authenticationStateChangeSubject = new Subject<boolean>();
```
Comment (Marvvvin5) [inline frontend/src/app/core/services/user.service.ts:14]: wow login finally

File: frontend/src/app/menu/menu.component.ts [hunk 5]
```diff
@@ -1,26 +1,46 @@
...
+import { UserService } from '../core/services/user.service';
 
 @Component({
   selector: 'app-menu',
   templateUrl: './menu.component.html',
   styleUrls: ['./menu.component.scss']
 })
 export class MenuComponent implements OnInit, OnDestroy {
-  isUserAuthenticated!: boolean;
+  isUserAuthenticated: boolean = false;
```
Comment (Marvvvin5) [inline frontend/src/app/menu/menu.component.ts]: we are getting it onInit so idk why specifing it as false

File: frontend/src/app/pages/login/login.component.ts [hunk 6]
```diff
@@ -1,10 +1,103 @@
...
+      return;
+    }
+
+    const userForLogin: UserForLoginDTO = {
+      email: this.loginForm.get('email')!.value,
+      password: this.loginForm.get('password')!.value
+    };
+
+    const login$ = this.userService.loginUser(userForLogin).pipe(
+      tap(() => {
```
Comment (Marvvvin5) [inline frontend/src/app/pages/login/login.component.ts:48]: if we are getting navigated to home page maybe we shouldnt display a message

File: frontend/test-object-models/login-object-model.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        const token = await this.page.evaluate(() => localStorage.getItem('token'));
+        expect(token != null).toBeTruthy();
+    }
+
+    async expectLoginFail() {
+        await expect(this.page).toHaveURL('http://localhost:4200/login');
+        const token = await this.page.evaluate(() => localStorage.getItem('token'));
+        expect(token).toBeNull();
+    }
+}
```
Comment (Marvvvin5) [inline frontend/test-object-models/login-object-model.ts:62]: what about expectedLogoutSuccess>

File: backend/YouTubeV2.Application/YTContext.cs (hunk 1 above)
Comment (KarolDenst) [reply backend/YouTubeV2.Application/YTContext.cs:22]: yes. we need them to give user  a role

File: frontend/src/app/pages/login/login.component.ts (hunk 6 above)
Comment (KarolDenst) [reply frontend/src/app/pages/login/login.component.ts:48]: I don't get your obsessionwith displaying messages

File: frontend/playwright.config.ts (hunk 3 above)
Comment (KarolDenst) [reply frontend/playwright.config.ts:1]: I is generated code with a small change. I commented firefox + safari. These are more restrictive and for our app to work on those we need to get a 3rd party authenticated cert or run everything on http. I will set it to work on http in the next commit but i don't know which option is better

Comment (KarolDenst) [reply frontend/playwright.config.ts:1]: im keeping https

File: frontend/src/app/core/services/user.service.ts [hunk 8]
```diff
@@ -22,4 +39,8 @@ export class UserService {
     
     return this.httpClient.get<UserDTO>(`${this.registrationPageWebAPIUrl}/user`, { params: params });
   }
+
+  loginUser(userForLogin: UserForLoginDTO): Observable<string> {
```
Comment (Marvvvin5) [inline frontend/src/app/core/services/user.service.ts:43]: idk if it shouldnt be Observable<void>. ask gpt hehe

**Review by Marvvvin5 — APPROVED** [review]

File: frontend/src/app/core/services/user.service.ts (hunk 8 above)
Comment (KarolDenst) [reply frontend/src/app/core/services/user.service.ts:43]: image
